scripts/install.sh

```diff
@@ -4,10 +4,39 @@ set -e
 
 os=$(uname | tr '[:upper:]' '[:lower:]')
 arch=$(uname -m | tr '[:upper:]' '[:lower:]' | sed -e s/x86_64/amd64/)
+if [ "$arch" = "aarch64" ]; then
+  arch="arm64"
+fi
+
+url="https://infracost.io/downloads/latest"
+tar="infracost-$os-$arch.tar.gz"
 echo "Downloading latest release of infracost-$os-$arch..."
-curl -sL https://github.com/infracost/infracost/releases/latest/download/infracost-$os-$arch.tar.gz | tar xz -C /tmp
+curl -sL $url/$tar -o /tmp/$tar
+echo
+
+code=$(curl -s -L -o /dev/null -w "%{http_code}" $url/$tar.sha256)
+if [ "$code" = "404" ]; then
+  echo "Checksum for infracost-$os-$arch release not found, skipping checksum validation"
+else
+  if [ -x "$(command -v shasum)" ]; then
+    echo "Validating checksum for infracost-$os-$arch..."
+    curl -sL $url/$tar.sha256 -o /tmp/$tar.sha256
+    (cd /tmp/;shasum -sqc $tar.sha256)
+    rm /tmp/$tar.sha256
+  else
+    echo "The shasum command could not be found, skipping checksum validation"
+  fi
+fi
+
+tar xzf /tmp/$tar -C /tmp
+rm /tmp/$tar
```

Comment:
Should we get rid of the `rm` commands? it could alarm users if the `rm` asks for confirmation, and we're deleting stuff in /tmp so it doesn't matter as it'll get cleaned up eventually if storage is low.

Reply:
Personally I'd prefer scripts to cleanup artefacts they created. It just seems good practice.

Reply:
tru dat! Overall it seems like we want this script to be non-interactive as people are copying that curl into their workflows, so if there is no risk of `rm` throwing a prompt then it's safe to ship.